src/main/java/com/atlassian/db/replica/internal/SQLOpeartionType.java

```diff
@@ -0,0 +1,50 @@
+package com.atlassian.db.replica.internal;
+
+public class SQLOpeartionType {
+
+    public static enum SQLType {
+        WRITE, READ_ONLY, NOT_KNOWN
+    }
+
+    private final SqlFunction sqlFunction;
+
+    public SQLOpeartionType(SqlFunction sqlFunction) {
+        this.sqlFunction = sqlFunction;
+    }
+
+    /**
+     * Arbitrially judge based on sql param whether sql is WRITE
+     */
+    public boolean whetherSQLisWrite(String sql){
+        //method to arbitrially judge whether sql isWrite or
+        //maybe inspired MaryPri
```

Comment:
MaryPri uses "is JQL", which is Jira-specific. It also uses a `org.postgresql.core.Parser`, which is Postgres-specific. Parsers are also a lot slower than String heuristics.

Reply:
There are libraries that are parsing SQLs. You'd need to cache results if it's slow.

Reply:
Caching won't help, we're parsing a different `sql` each time

Reply:
> There are libraries that are parsing SQLs.

https://github.com/atlassian-labs/db-replica/issues/9

Reply:
We are building better observability. If the current approach does not work well, then we can consider adding a lib. We need to take into account all the factors:
- parsing will be slower, caching may not help and will add more complexity,
- adding dependencies to a library reduces reusability.

Reply:
This is just an example. WIth parsing by external lib coupled with postgress. But it would be interesting to compare db-replica judgment(READ/WRITE) on Hello for 100% of queries with MaryPri and look for space for improvement

Reply:
In db-replica case the information about sql operation type should be known in compilation lvl as it operates on Statement & SQL low lvl internals.